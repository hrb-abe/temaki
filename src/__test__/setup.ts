import * as Enzyme from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import * as ContextHook from 'babel-plugin-require-context-hook/register'

Enzyme.configure({ adapter: new Adapter() })
ContextHook()
