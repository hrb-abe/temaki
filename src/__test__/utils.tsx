import { mount, ReactWrapper, shallow, ShallowWrapper } from 'enzyme'
import * as React from 'react'
import { defaultTheme, ThemeProvider } from '~/modules/theme'

// 推論で型エラーがでるので、型定義してキャストする
type ShallowWithTheme = (tree: React.ReactElement) => ShallowWrapper
export const shallowWithTheme: ShallowWithTheme = tree => {
  return shallow(<ThemeProvider theme={defaultTheme}>{tree}</ThemeProvider>)
}

// 推論で型エラーがでるので、型定義してキャストする
type MountWithTheme = (tree: React.ReactElement) => ReactWrapper
export const mountWithTheme: MountWithTheme = tree => {
  return mount(<ThemeProvider theme={defaultTheme}>{tree}</ThemeProvider>)
}
