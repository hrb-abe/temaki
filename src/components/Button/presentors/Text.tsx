import * as React from 'react'
import styled from '~/modules/theme'

import { buttonBaseMixin, rippleEffectMixin } from '../lib/styles'

/**
 * Component
 */

type Props = {
    iconSrc?: string
    onClick?: (e: React.MouseEvent<HTMLButtonElement>) => void
}
const Presentor: React.FC<Props> = ({ iconSrc, children, ...props }) => (
    <Outer {...props}>
        <Icon src={iconSrc} />
        <Text data-test="text-button-child">{children}</Text>
    </Outer>
)

/**
 * Styles
 */

const Outer = styled.button`
    ${buttonBaseMixin}
    ${rippleEffectMixin}
  height: 24px;
    width: auto;
`

const Icon = styled.img`
    width: 24px;
    height: 24px;
`

const Text = styled.span`
    color: ${props => props.theme.colors.text};
    font-size: 14px;
    font-weight: normal;
`

export default Presentor
