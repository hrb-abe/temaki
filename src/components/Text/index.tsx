import * as React from 'react'
import styled, { RequiredThemeProps } from '~/modules/theme'

// type Props = {
//   size?: number
//   level?: number
// }
const Text: React.FC = ({ children }) => {
  return <StyledText>{children}</StyledText>
}

const StyledText = styled.p`
  color: ${(props: { theme: RequiredThemeProps }) => props.theme.colors.text.default};
`

export default Text
