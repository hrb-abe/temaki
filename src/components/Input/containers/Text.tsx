import * as React from 'react'
import { ContainerType } from '~/types/utils'

import * as Input from '../index'

/**
 * Utils
 */

const useChangeStringValueFromChangeEvent = (
  onChange: ((value: Input.StringValue) => void) | undefined,
  onChangeNative: ((e: React.ChangeEvent<HTMLInputElement>) => void) | undefined
) =>
  React.useCallback(
    (e: React.ChangeEvent<HTMLInputElement>) => {
      e.persist()

      if (onChangeNative) {
        onChangeNative(e)
      }
      if (onChange) {
        onChange(e.target.value)
      }
    },
    [onChange]
  )

/**
 * Component
 */

type Props = {
  value?: Input.StringValue
  onChange?: (value: Input.StringValue) => void
  onChangeNative?: (e: React.ChangeEvent<HTMLInputElement>) => void
}
type InjectProps = {
  value?: Input.StringValue
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void
}
const Text: ContainerType<Props, InjectProps> = ({ Presenter, value, onChange, onChangeNative, ...props }) => {
  const changeValue = useChangeStringValueFromChangeEvent(onChange, onChangeNative)

  // @ts-ignore
  return <Presenter value={value} onChange={changeValue} {...props} />
}

export default Text
