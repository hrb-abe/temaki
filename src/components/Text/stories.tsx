import { text } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import * as React from 'react'
import Text from './index'

storiesOf('Text', module).add('Standard', () => <Text>{text('Value', 'Hello')}</Text>)
