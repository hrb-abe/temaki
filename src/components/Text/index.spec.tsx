import * as Enzyme from 'enzyme'
import toJson from 'enzyme-to-json'
import * as React from 'react'
import { mountWithTheme } from '~/__test__/utils'
import Text from './index'

describe('Text', () => {
  let wrapper: Enzyme.ReactWrapper

  beforeEach(() => {
    wrapper = mountWithTheme(<Text>Hello</Text>)
  })

  it('should be defined', () => {
    expect(wrapper).toBeDefined()
  })

  it('should render collectly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
