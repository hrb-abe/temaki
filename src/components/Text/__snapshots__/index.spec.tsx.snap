// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render collectly 1`] = `
<styled.p>
  Hello
</styled.p>
`;
